### PR TITLE
Fix(eos_designs): Correct underlay routing for overlay_routing_protocol: "none"

### LINKED ISSUE #1884: EVPN overlay variables generated although EVPN_role is none
### Issue Summary

tried creating l3 router node type and keys where the router will not be running EVPN. 

Defined the node type as :

```yaml
l3router:
    type: l3router
    connected_endpoints: true
    default_evpn_role: none
    mlag_support: true
    ip_addressing:
      mlag_ip_primary: 'templates/ip_addressing/mlag_ip_primary.j2'
      mlag_ip_secondary: 'templates/ip_addressing/mlag_ip_secondary.j2'
      mlag_l3_ip_primary: 'templates/ip_addressing/mlag_l3_ip_primary.j2'
      mlag_l3_ip_secondary: 'templates/ip_addressing/mlag_l3_ip_secondary.j2'    
    network_services: 
      l2: true
      l3: true
    underlay_router: true
    uplink_type: p2p  
```

So, the **evpn_role** is **None**.

End result which appeared wrong of the device type that is not running EVPN
 Name of the prefix list containing EVPN OVERLAY that could be deceiving to the user and does not seem like there is a way to override this name

```eos
ip prefix-list PL-LOOPBACKS-EVPN-OVERLAY
   seq 10 permit 2.2.2.0/24 eq 32
route-map RM-CONN-2-BGP permit 10
   match ip address prefix-list PL-LOOPBACKS-EVPN-OVERLAY
```

Also noticed the below under debug vars for this node and appeared in accurate as EVPN role is none:

```yaml
        overlay_address_families:
        - evpn
        overlay_routing_protocol: ebgp
        overlay_routing_protocol_address_family: ipv4
```

### Which component(s) of AVD impacted

eos_designs

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Correct underlay routing for overlay_routing_protocol: "none"

## Related Issue(s)

Recently we added support for `overlay_routing_protocol: "none"`. The PR did not include changes to underlay routing so the eos_designs behavior was to only redistribute the loopback IP into the underlay. This is wrong when not using an overlay, so instead we should redistribute all connected routes in default VRF to the underlay.

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

When `overlay_routing_protocol: "none"` redistribute all routes in default VRF to underlay routing protocol.

## How to test

Will be covered in future L2LS / L3LS molecule scenarios.

No changes to current molecule scenarios.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/templates/underlay/ebgp/router-bgp.j2 [hunk 1]
```diff
@@ -25,8 +25,12 @@ router_bgp:
         activate: true
 {% endif %}
   redistribute_routes:
+{% if switch.overlay_routing_protocol == "none" %}
+    connected: {}
+{% else %}
     connected:
       route_map: RM-CONN-2-BGP
+{% endif %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/templates/underlay/ebgp/router-bgp.j2:33]: Any reason, why we are not doing this way?

```suggestion
{% if switch.overlay_routing_protocol == "none" %}
  redistribute_routes:
    connected:
      route_map: RM-CONN-2-BGP
{% endif %}
```

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/underlay/ebgp/router-bgp.j2:33]: Those two are  not the same logic. Before this PR we had the redistribution of connected _with_ route-map. In the case on overlay_protocol: none I now want to remove the route_map.

**Review by onurgashi — APPROVED** [review]

File: ansible_collections/arista/avd/molecule/eos_designs-l2ls/intended/configs/BGP-SPINE1.cfg [hunk 2]
```diff
@@ -137,7 +131,7 @@ router bgp 65001
    neighbor 192.168.253.5 description DUMMY-CORE
    neighbor 192.168.254.1 peer group MLAG-IPv4-UNDERLAY-PEER
    neighbor 192.168.254.1 description BGP-SPINE2
-   redistribute connected route-map RM-CONN-2-BGP
+   redistribute connected
```
Comment (tgodaA) [inline ansible_collections/arista/avd/molecule/eos_designs-l2ls/intended/configs/BGP-SPINE1.cfg:134]: @ClausHolbechArista , I got what you said with this example.

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**